ui/src/map/plan-download/plan-download-store.ts

```diff
@@ -96,6 +100,33 @@ export const initialPlanDownloadStateFromSelection = (
     };
 };
 
+export const planDownloadAssetIdFromToolPanelAsset = (
+    selectedAsset: ToolPanelAsset,
+): PlanDownloadAssetId | undefined => {
+    switch (selectedAsset.type) {
+        case 'TRACK_NUMBER':
+            return {
+                type: PlanDownloadAssetType.TRACK_NUMBER,
+                id: brand(selectedAsset.id),
+            };
+        case 'LOCATION_TRACK':
+            return {
+                type: PlanDownloadAssetType.LOCATION_TRACK,
+                id: brand(selectedAsset.id),
+            };
+        case 'GEOMETRY_PLAN':
+        case 'GEOMETRY_ALIGNMENT':
+        case 'GEOMETRY_KM_POST':
+        case 'GEOMETRY_SWITCH':
+        case 'GEOMETRY_SWITCH_SUGGESTION':
+        case 'KM_POST':
+        case 'SWITCH':
+            return undefined;
+        default:
+            return exhaustiveMatchingGuard(selectedAsset.type);
+    }
+};
```

Comment:
Tämä on siis käytännössä `<SelectionPanelContainer>`:in sisällä aiemmin ollu `initialSelectionForPlanDownload()`. Samaa valintalogiikkaa käytetään myös silloin, kun popup avataan officialiinsiirtymisdialogin kautta. Funktio vaan näyttää hyvin erilaiselta kuin edeltäjänsä, sillä melkeinpä ainoa entiselleen jäänyt osuus siitä on ajatus funktion taustalla - funktion nimi meni täysin uusiksi, ja myös entinen if-else-rakenne vaihtui switch-caseen